ecl/hql/hqlfold.cpp

```diff
@@ -2154,6 +2154,82 @@ IHqlExpression * foldConstantOperator(IHqlExpression * expr, unsigned foldOption
             }
             break;
         }
+    case no_regex_findset:
+        {
+            IValue * t0 = expr->queryChild(0)->queryValue();
+            IValue * t1 = expr->queryChild(1)->queryValue();
+            IHqlExpression * c2 = queryRealChild(expr, 2);
+            IValue * t2 = c2 ? c2->queryValue() : NULL;
+            if (t0 && (!c2 || t2))
+            {
+                if(isUnicodeType(t0->queryType()))
+                {
+                    HqlExprArray results;
+                    unsigned plen = t0->queryType()->getStringLen();
+                    UChar * pattern = (UChar *)malloc((plen+1)*2);
+                    t0->getUCharStringValue(plen+1, pattern); //plen+1 so get null-terminated
+                    ICompiledUStrRegExpr * compiled = rtlCreateCompiledUStrRegExpr(pattern, !expr->hasAttribute(noCaseAtom));
+                    if (t1 && compiled)
+                    {
+                        unsigned slen = t1->queryType()->getStringLen();
+                        UChar * search = (UChar *)malloc((slen)*2);
+                        t1->getUCharStringValue(slen, search);
+
+                        IUStrRegExprFindInstance * match = compiled->find(search, 0, slen);
+                        while (match->found())
```

Comment:
The third parameter of getMatchX doesn't return the 'n'th match.  The third parameter specifies which part of the regular expression was matched.  For example, if you have the regular expression '([[:alpha:]]+)([[:digit:]+)' and the search string was 'Add 34 Subtract 45 Divide 4;, then the getMatchX(,,0) returns 'Add 34', getMatchX(,,1) returns 'Add' and getMatchX(,,2) returns '34'.  The size of the match (matchSize) would be 2.  To continue searching for further matches to match 'Subtract 45' and 'Divide 4', find() will need to be executed two more times.